pulse2percept/tests/test_files.py

```diff
@@ -3,11 +3,197 @@
+    npt.assert_equal(video.shape, [250, 272, 640, 1])
+
+    # Load as TimeSeries
+    video = files.load_video(datasets.bikes(), as_timeseries=True)
+    fps = files.load_video_framerate(datasets.bikes())
+    npt.assert_equal(isinstance(video, utils.TimeSeries), True)
+    npt.assert_almost_equal(video.tsample, 1.0 / fps)
+    npt.assert_equal(video.shape, [272, 640, 3, 250])
+
+    # Trigger an import error
+    with mock.patch.dict("sys.modules", {"skvideo": {}, "skvideo.utils": {}}):
+        with pytest.raises(ImportError):
+            reload(files)
+            files.load_video('invalid.avi')
+
+
+def test_load_video_generator():
+    # Load a test example
+    reload(files)
+
+    from skvideo import datasets
+    reader = files.load_video_generator(datasets.bikes())
+    for frame in reader.nextFrame():
+        npt.assert_equal(frame.shape, [272, 640, 3])
+
+    # Trigger an import error
+    with mock.patch.dict("sys.modules", {"skvideo": {}, "skvideo.utils": {}}):
+        with pytest.raises(ImportError):
+            reload(files)
+            files.load_video_generator('invalid.avi')
+
+
+def test_save_video():
+    # Load a test example
+    reload(files)
+    from skvideo import datasets
+
+    # There and back again: ndarray
+    videoin = files.load_video(datasets.bikes())
+    fpsin = files.load_video_framerate(datasets.bikes())
+    files.save_video(videoin, 'myvideo.mp4', fps=fpsin)
+    videout = files.load_video('myvideo.mp4')
+    npt.assert_equal(videoin.shape, videout.shape)
+    npt.assert_almost_equal(videout / 255.0, videoin / 255.0, decimal=0)
+
+    # Write to file with different frame rate, widths, and heights
+    fpsout = 15
+    files.save_video(videoin, 'myvideo.mp4', width=100, fps=fpsout)
+    npt.assert_equal(files.load_video_framerate('myvideo.mp4'), fpsout)
+    videout = files.load_video('myvideo.mp4')
+    npt.assert_equal(videout.shape[2], 100)
+    files.save_video(videoin, 'myvideo.mp4', height=20, fps=fpsout)
+    videout = files.load_video('myvideo.mp4')
+    npt.assert_equal(videout.shape[1], 20)
+    videout = None
+
+    # There and back again: TimeSeries
+    tsamplein = 1.0 / float(fpsin)
+    tsampleout = 1.0 / float(fpsout)
+    rollaxes = np.roll(range(videoin.ndim), -1)
+    tsin = utils.TimeSeries(tsamplein, np.transpose(videoin, rollaxes))
+    files.save_video(tsin, 'myvideo.mp4', fps=fpsout)
+    npt.assert_equal(tsin.tsample, tsamplein)
+    tsout = files.load_video('myvideo.mp4', as_timeseries=True)
+    npt.assert_equal(files.load_video_framerate('myvideo.mp4'), fpsout)
+    npt.assert_equal(isinstance(tsout, utils.TimeSeries), True)
+    npt.assert_almost_equal(tsout.tsample, tsampleout)
+
+    # Also verify the actual data
+    tsres = tsin.resample(tsampleout)
+    tsin = None
+    npt.assert_equal(tsout.shape, tsres.shape)
+    npt.assert_almost_equal(tsout.data / 255.0, tsres.data / tsres.data.max(),
+                            decimal=0)
+    tsres = None
```

Comment:
Currently it produces a warning and ignores the layer. But I can upgrade it to an error.